taichi/transforms/scalarize.cpp

```diff
@@ -540,7 +549,8 @@ class ScalarizePointers : public BasicStmtVisitor {
       stmt->replace_all_usages_with(scalarized_alloca_stmt)
   */
   void visit(MatrixPtrStmt *stmt) override {
-    if (stmt->origin->is<AllocaStmt>()) {
+    if (stmt->origin->is<AllocaStmt>() &&
+        is_alloca_scalarizable(stmt->origin->cast<AllocaStmt>())) {
       auto alloca_stmt = stmt->origin->cast<AllocaStmt>();
       auto tensor_type =
           alloca_stmt->ret_type.ptr_removed()->cast<TensorType>();
```

Comment:
```suggestion
    if (stmt->origin->is<AllocaStmt>()) {
      auto alloca_stmt = stmt->origin->cast<AllocaStmt>();
      auto tensor_type =
          alloca_stmt->ret_type.ptr_removed()->cast<TensorType>();
      TI_ASSERT(tensor_type != nullptr);
      if (is_alloca_scalarizable(alloca_stmt)) {
```
The next line (`if (tensor_type) {`) needs to be removed as well. GitHub doesn't allow me to do so :-)